client-tags/message-status.md

```diff
@@ -0,0 +1,83 @@
+---
+title: IRCv3 `message-status` client tags
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "evan@muffinmedic.net"
+---
+## Purpose
+This specification defines several `message-status` client tags, which allows clients to send and receive various information about the status of messages before and after they are sent.
+
+## Description
+The specification includes three client tags for message tracking: `typing`, `delivered`, and `read`. The `typing` tag indicates that a user is currently typing a message to a specific target. `delivered` and `read` provide for delivery and read receipts, respectively, and enable users to know when and by whom a message has been delivered to or read.
+
+For the purpose of this specification, the terms "delivery receipt" and "read receipt" are synonymous with the use of the `delievered` and `read` client tags, respectively.
+
+## Format
+All tags are sent as a [`TAGMSG`][tags] with an appropriate [`server-time`][time]. `delivered` and `read` must also include the [`labeled-response` tag][labels] of the original message. `typing` can be either `true` or `false`; `delivered` and `read` contain no values.
+
+### `typing`
+The `typing=true` client tag SHOULD be sent by the client when the first character is typed in the text-input field. Upon receipt of the `PRIVMSG` or end of a multiline [`batch`][batch], clients SHOULD assume that `typing=false` (no explicit tag needs to be sent).
+
+If new text is entered into the text-input field following sending of the `PRIVMSG` or [`batch`][batch], the client MUST send a new `typing=true` tag after sending the preceding `PRIVMSG` or end of [`batch`][batch].
+
+Upon clearing of the text-input field by the user (e.g., message cancelled), the client MUST send a `typing=false` tag to notify other users.
+
+After receiving a `typing=true` tag, the client SHOULD assume the state is unchanged until either a `PRIVMSG`,[`batch`][batch] end, or `typing=false` is received.
+
```

Comment:
This would work better if `typing` was a valueless tag that gets sent every few seconds and receiving clients are instructed to unset the status after a timeout of the same number of seconds + some slop.

Removes some complexity from the sending client of having to track typing status, instead of simply sending a debounced "yup still typing" announcement. Also, prevents the case of a stuck typing indicator if the sending client gets disconnected before they can send a `typing=false`.